core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -246,6 +246,46 @@ class ExecutorSuite extends SparkFunSuite
     heartbeatZeroAccumulatorUpdateTest(false)
   }
 
+  test("SPARK-19147: Gracefully handle error in task after executor is stopped") {
```

Comment:
The purpose of the change is really to avoid stack traces, not assert about the particular error. Would this pass even before this change? I'm just wondering if this is worth testing.

What may be worth testing is whether metrics are updated, which is the real possible behavior change here? or would they already be reported in case of an error?

Reply:
Remove the case after I make clearly how metrics works, thanks for review.